src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -89,37 +100,34 @@ private static int GetInputBuffer(ref StringBuffer content, bool isDosUnc, ref S
                 : PathInternal.DevicePrefixLength;
 
             buffer.EnsureCapacity(length + 1);
+            buffer.Length = 0;
 
             if (isDosUnc)
             {
-                // Put the extended UNC prefix (\\?\UNC\) in front of the path
-                buffer.CopyFrom(bufferIndex: 0, source: PathInternal.UncExtendedPathPrefix);
+                // Is a \\Server\Share, put \\?\UNC\ in the front
+                buffer.Append(PathInternal.UncExtendedPathPrefix);
 
-                // Copy the source buffer over after the existing UNC prefix
-                content.CopyTo(
-                    bufferIndex: PathInternal.UncPrefixLength,
-                    destination: ref buffer,
-                    destinationIndex: PathInternal.UncExtendedPrefixLength,
-                    count: content.Length - PathInternal.UncPrefixLength);
+                // Copy Server\Share\... over to the buffer
+                buffer.Append(buffer.AsSpan().Slice(PathInternal.UncPrefixLength));
```

Comment:
This does not look right. The buffer is only going to have `UncExtendedPathPrefix` in it at this point, and so the Slice will be alway empty.

Reply:
Should it be `content`?

Reply:
Yes, good catch. Apparently I'm missing a test case here (UNC with short name).